components/SearchInput.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useRef } from "react";
+
+import Input from "./form/input";
+
+export default function SearchInput({ value, onChange }) {
+  const inputRef = useRef();
+
+  // Makes autofocus work on search input
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+
+  return (
+    <Input
+      placeholder="Search user by name or tags, e.g., open source, reactjs"
+      inputRef={inputRef}
+      name="keyword"
+      value={value}
+      onChange={(e) => onChange(e.target.value)}
+    />
+  );
+}
```

Comment:
I don't think this file is needed, seems like too much abstraction to me as we are unlikely to reuse this?

Reply:
Earlier this was required since there was no Input component. Now, I think there is only one difference which is autofocus on the Input box. Do you suggest I should put the autofocus as a boolean prop on the `Input` component and use?

Reply:
But the `Input` component has `restProps` so it should be ok I think?

Reply:
No, the `Input` component will have its own `ref` and I will do the autofocus based on the boolean prop inside the `useEffect`. That's how it will work if we want to use a prop